### DISCUSSION TITLE
Voice navigation per junction, not per distance?

### DISCUSSION THREAD

**Discussion by dl-bone** [osmandapp/OsmAnd#13978] [Q&A]
Is it possible to get the voice navigation to tell you to "turn right at next junction" ("or take the next street right" or sth. like that) instead of "turn right in 150 meters"? 
I've been cycling around a lot without a phone holder on my bike, and frequently missed turns, especially when there was much traffic and I couldn't concentrate on the nav voice.
A real co-driver would probably also say something like "turn right after next tunnel / POI", but I guess this would be hard to get right.

There are some other issues I've encountered, but I'm not sure if these are from osmand or osm..
For example, if a road gets two physically separated lanes after a junction[1], I would expect to hear a "follow road" or something like that. Instead it is a "turn right then immediately left". 
The same happens when a designated bike lane begins or ends, and I get navigated on or off the main road. 
Or when a junction has a traffic island[2], the instruction is "turn left and then immediately left" instead of just "turn left". Is there a way to change that?

I've used the German, German(informal) and English TTS, and they don't seem to differ much.

[1] a road turning into two separated lanes.
![Screenshot_20220323-191424](https://user-images.githubusercontent.com/102251436/159769999-0d6a92ef-1746-47e4-9565-2c06e110b6b7.png)

[2] a traffic island.
![Screenshot_20220323-192342](https://user-images.githubusercontent.com/102251436/159770008-8b980c23-a7b9-4170-85c9-a4eeba3a9eb8.png)

**Comment by mikehgentry** [comment]
You could definitely edit the tts (text to speech) js files to get rid of the distance announcements if you wanted. Problem is it's a bit unpredictable when it fires off an instruction, so you'd probably want to put a limit on how far out it'll do them (so it doesn't say "turn right at the next junction" when it's 2km away). I opted for beeps instead of using speech at all, and it works reasonably well. There's a few hints in the thread:

https://github.com/osmandapp/OsmAnd/issues/10501
